metaseq/data/jsonl_dataset.py

```diff
@@ -88,6 +88,13 @@ def _build_index(self, path: str):
         cur = 0
         while True:
             line = f.readline()
+            if line != b"":
+                try:
+                    json.loads(line)
+                except json.decoder.JSONDecodeError:
+                    raise AssertionError(
+                        "Unable to load truncated or corrupted JSON file"
```

Comment:
Instead of an assertion error, let's keep with JSONDecodeError but provide a better error message. Include the name of the file and the line number in the error message, please

Reply:
Noted. Updated the file to do this.